superbench/common/utils/topo_aware.py

```diff
@@ -0,0 +1,211 @@
+
+import re
+import networkx as nx
+
+
+class quick_regexp(object):
+    """
+    Quick regular expression class
+
+    """
+
+    def __init__(self):
+        self.groups = None
+        self.matched = False
+
+    def search(self, pattern, string, flags=0):
+        match = re.search(pattern, string, flags)
+        if match:
+            self.matched = True
+            if match.groups():
+                self.groups = re.search(pattern, string, flags).groups()
+            else:
+                self.groups = True
+        else:
+            self.matched = False
+            self.groups = None
+
+        return self.matched
+
+
+def gen_topo_aware_config(hosts, ibstat_file, topology_file, config_path):
```

Comment:
shall we add some unit tests for this util function? e.g., add some fake ibstat and ibnetdiscover files and test those files

Reply:
Added Unit Test in new rev. 
The added `tests/data/ib_traffic_topo_aware_expected_config` has been verified through three algorithms. (all_pairs_shortest_path_length, all_pairs_dijkstra_path_length, all_pairs_bellman_ford_path_length) in networkx package to make sure it's correct.